src/main/java/school/faang/user_service/service/subscription/SubscriptionServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package school.faang.user_service.service.subscription;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import school.faang.user_service.dto.subscription.SubscriptionUserDto;
+import school.faang.user_service.dto.subscription.UserFilterDto;
+import school.faang.user_service.entity.User;
+import school.faang.user_service.mapper.UserMapper;
+import school.faang.user_service.repository.SubscriptionRepository;
+import school.faang.user_service.service.subscription.filters.UserFiltersApplier;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class SubscriptionServiceImpl implements SubscriptionService {
+    private final SubscriptionRepository subscriptionRepository;
+    private final UserFiltersApplier userFilter;
+    private final UserMapper userMapper;
+    private final SubscriptionValidator validator;
+
+    @Override
+    @Transactional
+    public void followUser(Long followerId, Long followeeId) {
+        validator.checkIfSubscriptionToHimself(followerId, followeeId);
+        validator.checkIfSubscriptionNotExists(followerId, followeeId);
+        validator.validateUserIds(followerId, followeeId);
+        subscriptionRepository.followUser(followerId, followeeId);
+    }
+
+    @Override
+    @Transactional
+    public void unfollowUser(Long followerId, Long followeeId) {
+        validator.checkIfSubscriptionExists(followerId, followeeId);
+        subscriptionRepository.unfollowUser(followerId, followeeId);
+    }
+
+    @Override
+    @Transactional
```

Comment:
Здесь тоже без транзакции ошибка? Вроде модифицирующих запросов к базе данных нет

Reply:
Да тут тоже, причем getCount без аннотации работает а getFollowers и getFollowing нет

Reply:
2024-09-11 05:33:33.179 [http-nio-8080-exec-2] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: org.springframework.dao.InvalidDataAccessApiUsageException: You're trying to execute a streaming query method without a surrounding transaction that keeps the connection open so that the Stream can actually be consumed; Make sure the code consuming the stream uses @Transactional or any other way of declaring a (read-only) transaction] with root cause
org.springframework.dao.InvalidDataAccessApiUsageException: You're trying to execute a streaming query method without a surrounding transaction that keeps the connection open so that the Stream can actually be consumed; Make sure the code consuming the stream uses @Transactional or any other way of declaring a (read-only) transaction

Reply:
Может потому что stream возвращается из репозитория?

Reply:
Да, точно)